src/crawlers/statistics.js

```diff
@@ -100,20 +154,118 @@ export default class Statistics {
         };
     }
 
-    startLogging() {
+    /**
+     * Initializes the key value store for persisting the statistics,
+     * displaying the current state in predefined intervals
+     */
+    async startLogging() {
+        this.keyValueStore = await openKeyValueStore();
+
+        await this._maybeLoadStatistics();
+
+        this._listener = this.persistState.bind(this);
+
+        events.on(ACTOR_EVENT_NAMES_EX.PERSIST_STATE, this._listener);
```

Comment:
`BasicCrawler` pauses on migration, but allows some time for running requests to finish. See `_pauseOnMigration`. If you persist statistics immediately on the event, you'll lose information about the subsequently finished requests.

It's not ideal and the whole `events` integration would deserve a review, but it is what we have to work with now.

Reply:
the persistState passes the `isMigrating` status to the event listener, would suffice to not try to persist when this is true?

Reply:
I guess you can keep persisting in both cases. If you try to persist after the pause, the container could already have its network connectivity killed and you would not persist anything. So I think the safest bet is to always persist on `persistState` and then attempt to persist again after `pause`.